platforms/software/build-init/src/test/groovy/org/gradle/api/tasks/wrapper/WrapperTest.groovy

```diff
@@ -16,39 +16,48 @@
 
 package org.gradle.api.tasks.wrapper
 
+
 import org.gradle.api.tasks.AbstractTaskTest
 import org.gradle.api.tasks.TaskPropertyTestUtils
-import org.gradle.api.tasks.wrapper.internal.DefaultWrapperVersionsResources
+import org.gradle.integtests.fixtures.TestResources
 import org.gradle.test.fixtures.file.TestFile
+import org.gradle.test.fixtures.keystore.TestKeyStore
+import org.gradle.test.fixtures.server.http.BlockingHttpsServer
 import org.gradle.util.GradleVersion
 import org.gradle.util.internal.GUtil
 import org.gradle.util.internal.WrapUtil
 import org.gradle.wrapper.GradleWrapperMain
 import org.gradle.wrapper.WrapperExecutor
+import org.junit.Rule
 
 class WrapperTest extends AbstractTaskTest {
-    private static final String RELEASE = "7.6"
-    private static final String RELEASE_CANDIDATE = "7.7-rc-5"
-    private static final String NIGHTLY = "8.1-20221207164726+0000"
-    private static final String RELEASE_NIGHTLY = "8.0-20221207164726+0000"
+
```

Comment:
🤔 It would be nice to have a test to verify the behavior (and suggestions that appear) when upgrading to a nonexistent version.